src/codemagic/tools/_app_store_connect/arguments.py

```diff
@@ -96,6 +99,91 @@ class AppStoreConnectArgument(cli.Argument):
     )
 
 
+class BuildArgument(cli.Argument):
+    APPLICATION_ID = cli.ArgumentProperties(
+        key='application_id',
+        description='Application Apple ID',
+    )
+    APPLICATION_ID_OPTIONAL = cli.ArgumentProperties(
+        key='application_id',
+        flags=('--application-id',),
+        description='Application Apple ID',
+        argparse_kwargs={'required': False},
+    )
+    EXPIRED = cli.ArgumentProperties(
+        key='expired',
+        flags=('--expired',),
+        type=bool,
+        description='List only expired builds',
+        argparse_kwargs={
+            'required': False,
+            'action': 'store_true',
+        },
+    )
+    NOT_EXPIRED = cli.ArgumentProperties(
+        key='not_expired',
+        flags=('--not-expired',),
+        type=bool,
+        description='List only not expired builds',
+        argparse_kwargs={
+            'required': False,
+            'action': 'store_true',
+        },
+    )
+    BUILD_ID = cli.ArgumentProperties(
+        key='build_id',
+        description='Specific build ID',
+    )
+    BUILD_ID_OPTIONAL = cli.ArgumentProperties(
+        key='build_id',
+        flags=('--build-id',),
+        description='Specific build ID',
+        argparse_kwargs={'required': False},
+    )
+    PRE_RELEASE_VERSION= cli.ArgumentProperties(
+        key='pre_release_version',
+        flags=('--pre-release-version',),
+        description='Pre-release version of your application',
+        argparse_kwargs={'required': False},
+    )
+    PROCESSING_STATE = cli.ArgumentProperties(
+        key='processing_state',
+        flags=('--processing-state',),
+        type=BuildProcessingState,
+        description='Build processing state',
+        argparse_kwargs={
+            'required': False,
+            'choices': list(BuildProcessingState),
+        },
+    )
+    BUILD_NUMBER = cli.ArgumentProperties(
+        key='build_number',
+        flags=('--build-number',),
+        description='Build number',
+        argparse_kwargs={'required': False},
+    )
+    ORDERING = cli.ArgumentProperties(
```

Comment:
don't know what's needed:) as for me, i don't know a real use case for this action because it doesn't seamlessly provide other resources like apps or versions. Deleting order won't hurt

Reply:
I think just remove it for now from here and all related places. Later we can add more generic ordering support for all listing actions. It should make many things simpler.